src/shogun/classifier/AveragedPerceptron.cpp

```diff
@@ -91,5 +92,9 @@ void CAveragedPerceptron::iteration()
 		linalg::update_mean(bias, cached_bias, num_prev_weights);
 		num_prev_weights++;
 	}
+
+	observe(ObservedValue::make_observation(m_current_iteration, "w", "Model weights", w));
```

Comment:
Without `put()` we would need to do it the "old way".

Reply:
I see. Well that would be the thing for xvalidation then, but definitely not for the weights of the perceptron (those should go via put)